### PR TITLE
Check if email is used when updating user (#21289)

### LINKED ISSUE #21075: EditUserPost doesn't check email
### Description

When creating a new user I can't create a user if there is a user with the same email.

When I'm editing user in admin page, I can change email to existing one. 

### Gitea Version

1.17.1

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Using compose file

```yaml
---
version: "2"

services:
  db:
    image: bitnami/postgresql:latest
    user: 1001
    environment:
      - POSTGRESQL_USER=gitea
      - POSTGRESQL_PASSWORD=gitea
      - POSTGRESQL_DATABASE=gitea
    networks:
      - gitea-network

  gitea:
    image: gitea/gitea:1.17.1-rootless
    user: 1000:1000
    environment:
      - DB_TYPE=postgres
      - DB_HOST=db:5432
      - DB_NAME=gitea
      - DB_USER=gitea
      - DB_PASSWD=gitea
    volumes:
      - ./gitea-config:/etc/gitea
      - ./gitea-data:/var/lib/gitea
    ports:
      - "3000:3000"
    depends_on:
      - db
    networks:
      - gitea-network

  ldap1:
    image: glauth/glauth
    user: 1000
    volumes:
      - ./ldap-config.cfg:/app/config/config.cfg
    networks:
      - gitea-network
```

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by ghost**
Backport #21289

Fix #21075
When updating user data should check if email is used by other users

### CODE REVIEW

**Review by wolfogre — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

File: models/user/user.go [hunk 1]
```diff
@@ -891,14 +891,19 @@ func UpdateUser(ctx context.Context, u *User, changePrimaryEmail bool, cols ...s
 		if err != nil {
 			return err
 		}
-		if !has {
-			// 1. Update old primary email
-			if _, err = e.Where("uid=? AND is_primary=?", u.ID, true).Cols("is_primary").Update(&EmailAddress{
-				IsPrimary: false,
-			}); err != nil {
-				return err
+		if has && emailAddress.UID != u.ID {
```
Comment (lafriks) [inline models/user/user.go]: What if email exists and is not active? This way I could add to my account let's say others person's email address and block him from adding his other email

**Review by lafriks — CHANGES_REQUESTED** [review]

File: models/user/user.go (hunk 1 above)
Comment (delvh) [reply models/user/user.go]: Agreed, but that brings another change with it:
We need to throw out unactivated emails then when someone else changes his email and verifies it.

Comment (ghost) [reply models/user/user.go]: It's a backport, should I modify this patch?

Comment (lafriks) [reply models/user/user.go]: Yes, I think this should be fixed

**Comment by xin-u** [comment]
> CI failure is related.

wow, just noticed.

**Comment by jolheiser** [comment]
Going to close this as we've moved on from `1.17.x`

**Pull request closed without merging**